matrix/basic_matrix.py

```diff
@@ -144,15 +154,58 @@ def diagonal(self):
             result.append(self.ls_entries[i][i])
         return result
 
+    def LU_decomp(self):
+        """
+        Returns: the LU decomposition of the self.ls_entries matrix
+        self.ls_entries matrix must be square
+            L, U where [A] = [L][U]
+            L: <Matrix> m x m lower triangular matrix
+            U: <Matrix> m x m upper triangular matrix
+        """
+        # check if square
+        if not self.is_square():
+            raise NotImplemented('LU factorization for non-square matrices is not implemented.')
+
+        # L and U are both square matrices
+        L = self.identity(self.len_row).ls_entries
+        U = [[0] * self.len_row for x in range(self.len_row)]
+        A = copy.deepcopy(self.ls_entries)
+
+        # Need to pivot first for stability
+        for i in range(self.len_row):
+            minor_col_i = [abs(x[i]) for k, x in enumerate(A) if k >= i]
```

Comment:
I was thinking about this a bit more, and if we want to be able to use the same syntax as a numpy array then we can simply define that in the class, so it would just be us adding the class method `__getitem__(self, key)` and then we would be able to do `self[row][col]`

Reply:
I like that!  I added that into the basic_matrix class